Gems/GradientSignal/Code/Source/Components/ImageGradientComponent.cpp

```diff
@@ -431,17 +449,16 @@ namespace GradientSignal
                 // UVs outside the 0-1 range are treated as infinitely tiling, so that we behave the same as the 
                 // other gradient generators.  As mentioned above, if clamping is desired, we expect it to be applied
                 // outside of this function.
-                auto x = aznumeric_cast<AZ::u32>(pixelLookup.GetX()) % width;
-                auto y = aznumeric_cast<AZ::u32>(pixelLookup.GetY()) % height;
+                float pixelX = pixelLookup.GetX();
+                float pixelY = pixelLookup.GetY();
+                auto x = aznumeric_cast<AZ::u32>(pixelX) % width;
+                auto y = aznumeric_cast<AZ::u32>(pixelY) % height;
 
-                // Flip the y because images are stored in reverse of our world axes
-                y = (height - 1) - y;
+                // Retrieve the pixel value from the image data
+                float value = GetPixelValue(x, y);
 
-                // For terrarium, there is a separate algorithm for retrieving the value
-                const float value = (m_currentChannel == ChannelToUse::Terrarium)
-                    ? GetTerrariumPixelValue(x, y)
-                    : AZ::RPI::GetImageDataPixelValue<float>(
-                        m_imageData, m_imageDescriptor, x, y, aznumeric_cast<AZ::u8>(m_currentChannel));
+                // Handle any additional processing based on our sampling type
+                HandleSamplingType(value, x, y, pixelX, pixelY);
```

Comment:
The logic/math would be easier to follow if this first GetPixelValue() call got moved into HandleSamplingType, so that each case statement in HandleSamplingType did all the work it needed for calculating the final value instead of splitting it like this. It was confusing not seeing Point sampling explicitly handled in HandleSamplingType. :(

Reply:
OK I can move it. I was trying to optimize for the default case (point), but I can see how it is confusing this way.